tests/test_plugin.py

```diff
@@ -2,10 +2,106 @@
 
 import pytest
 from duty.tools import mkdocs
+from mkdocs.config.defaults import MkDocsConfig
+from mkdocs.structure.files import Files
+from mkdocs.structure.pages import Page
+from types import SimpleNamespace
+
+from mkdocs_llmstxt._internal.plugin import MkdocsLLMsTxtPlugin
 
 
 def test_plugin() -> None:
     """Run the plugin."""
     with pytest.raises(expected_exception=SystemExit) as exc:
         mkdocs.build()()
     assert exc.value.code == 0
+
+
+def test_page_descriptions() -> None:
+    """Test that page descriptions are correctly handled and included in output."""
+    # Create a mock config
+    config = MkDocsConfig()
+    config.site_name = "Test Project"
+    config.site_description = "Test Description"
+    config.site_url = "https://test.com/"
+    
+    # Create plugin instance with test configuration
+    plugin = MkdocsLLMsTxtPlugin()
+    plugin.load_config({
+        "sections": {
+            "Test Section": {
+                "page1.md": "Description of page 1",
+                "page2.md": "Description of page 2"
+            }
+        }
+    })
+    
+    # Initialize plugin
+    plugin.on_config(config)
+    
+    # Create mock file and page
+    file = SimpleNamespace(
+        src_uri="page1.md",
+        dest_uri="page1.html",
+        abs_dest_path="/tmp/page1.html",
+        url="page1.html"
+    )
+    page = Page(
+        title="Test Page",
+        file=file,
+        config=config
+    )
+    
+    # Process page content
+    plugin.on_page_content("<html><body>Test content</body></html>", page=page)
+    
+    # Check that the page info was stored correctly
+    assert len(plugin.md_pages["Test Section"]) == 1
+    page_info = plugin.md_pages["Test Section"][0]
+    assert page_info.title == "Test Page"
+    assert page_info.description == "Description of page 1"
+
+
+def test_mixed_descriptions() -> None:
+    """Test that mixing files with and without descriptions works correctly."""
+    # Create a mock config
+    config = MkDocsConfig()
+    config.site_name = "Test Project"
+    config.site_url = "https://test.com/"
+    
+    # Create plugin instance with test configuration
+    plugin = MkdocsLLMsTxtPlugin()
+    plugin.load_config({
+        "sections": {
+            "Test Section": {
+                "page1.md": "Description of page 1",
+                "page2.md": None,
```

Comment:
oh, maybe I take back my previous comments if yaml will turn this value-less key into a key with null. TIL